search/tutorials/tutorial_ordering.md

```diff
@@ -1,136 +1,216 @@
 
-Run the sample in a terminal using the following command:
-```bash
-python search_with_ordering.py
-```
+1. Run the code sample and check the Retail API in action.
+
+**Note**: Click the copy button on the side of the code box to paste the command in the Cloud Shell terminal and run it.
+
+### Set the PROJECT_NUMBER environment variable
+
+Because you are going to run the code samples in your own Google Cloud project, you should specify the **project_number** as an environment variable. It will be used in every request to the Retail API.
+
+1. You can find the ```project_number``` in the **Home/Dashboard/Project Info card**.
+
+1. Set the environment variable with the following command:
+    ```bash
+    export PROJECT_NUMBER=<YOUR_PROJECT_NUMBER>
+    ```
+
+### Install Google Cloud Retail libraries
+
+To run Python code samples for the Retail API tutorial, you need to set up your virtual environment.
+
+1. Run the following commands in a Terminal to create an isolated Python environment:
+    ```bash
+    pip install virtualenv
+    virtualenv myenv
+    source myenv/bin/activate
+    ```
+1. Next, install Google packages:
+    ```bash
+    pip install google
+    pip install google-cloud-retail
+    ```
+
+## Clone the Retail code samples (optional)
+
+This step is required if this is the first Retail API Tutorial you run.
+Otherwise, you can skip it.
+
+Clone the Git repository with all the code samples to learn the Retail features and check them in action.
+
+<!-- TODO(ianan): change the repository link -->
+1. Run the following command in the Terminal:
+    ```bash
+    git clone https://github.com/t-karasova/grs-samples-python.git
+    ```
 
-As you can see now, ```results[]``` are ordered by price descending.
+    The code samples for each of the Retail services are stored in different directories.
 
-## Ordering by a single field. Product sorting
+1. Go to the ```grs-samples-python``` directory. It's our starting point to run more commands.
+    ```bash
+    cd grs-samples-python
+    ```
 
-Next, let's change the ordering direction to show the cheapest products first.
+## Import catalog data (optional)
 
-To do that, find the comment 
+This step is required if this is the first Retail API Tutorial you run.
+Otherwise, you can skip it.
 
-"#TRY DIFFERENT ORDERING EXPRESSIONS HERE:" 
+### Upload catalog data to Cloud Storage
 
-and change the ordering expression to the following one:
+We have prepared a JSON file with valid products in the `search` directory:
 
-```order = 'price asc'``` or just ```order = 'price'``` - those are equivalent expressions since ascending is the default ordering direction.
+**search/products_for_search.json**
 
-Run the sample in a terminal using the command:
+You can use this file in the tutorial.
+
+1.  In your own Google Cloud project, go to Cloud Storage.
+1.  Click **Create bucket**, give your Project ID in the name, and
```

Comment:
1. Rearrange the numbered list.
2. "give your Project ID in the name" - what does this mean? Provide your PROJECT_ID in the name field?